### ISSUE TITLE
[vcpkg] toolsrc CMake broken build since adding x-set-installed command

### ISSUE THREAD

**Issue by linquize** [microsoft/vcpkg#10957]
since 556325a1f7b6049d91565257c00db2f0bf1eadc5 (adding x-set-installed command)
vcpkg toolsrc CMake broken build with MSVC cmake ninja.

Open x64 Native Tools Command Prompt for VS2019
```bat
cd C:\vcpkg\toolsrc
mkdir build
cd build
cmake .. -G Ninja
cmake --build .
```

```
[22/96] Building CXX object CMakeFiles...kglib.dir\src\vcpkg\base\files.cpp.obj
FAILED: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.obj
C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1425~1.286\bin\Hostx64\x64\cl.exe  /nologo /TP -DVCPKG_DISABLE_METRICS=0 -DVCPKG_USE_STD_FILESYSTEM=1 -I..\include /DWIN32 /D_WINDOWS  /GR /EHsc /MDd /Zi /Ob0 /Od /RTC1   -FC -permissive- -utf-8 -W4 -WX /Yupch.h /FIpch.h /Zm200 -std:c++17 /showIncludes /FoCMakeFiles\vcpkglib.dir\src\vcpkg\base\files.cpp.obj /FdCMakeFiles\vcpkglib.dir\ /FS -c ..\src\vcpkg\base\files.cpp
C:\vcpkg\toolsrc\src\vcpkg\base\files.cpp(725): error C2039: 'system_complete': is not a member of 'std::filesystem'
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\fstream(28): note: see declaration of 'std::filesystem'
C:\vcpkg\toolsrc\src\vcpkg\base\files.cpp(725): error C3861: 'system_complete': identifier not found
[27/96] Building CXX object CMakeFiles...pkglib.dir\src\vcpkg\base\json.cpp.obj
ninja: build stopped: subcommand failed.
```

`toolsrc/src/vcpkg/base/files.cpp` should use `VCPKG_USE_STD_FILESYSTEM` instead of `USE_STD_FILESYSTEM`

**Comment by NancyLi1013** [comment]
Hi @linquize 

Thanks for posting this issue.

Confirmed, this should be a vcpkg bug.
We will try to fix this soon.

**Comment by strega-nil** [comment]
Fixed in #10973 :)
